Automattic/wp-calypso
# Store: Settings: Warn of Unsaved Changes if Navigating

Partially addresses #17271 (Payments and Taxes only)

WORK IN PROGRESS

To test:
```
* Navigate to Store > Settings > Payments
* Change your currency setting but don't save the change
* Click on a sidebar item (e.g. Reviews)
* Verify you are prompted that you have unsaved changes
* Hit Cancel

* Put the currency setting back the way it was
* Click on a sidebar item (e.g. Reviews)
* Verify you are NOT prompted that you have unsaved changes

* Change another payments setting, e.g. Your PayPal ID (email address for PayPal), Title for Check Payments, Test Secret Key for Stripe, etc etc.
* Click on a sidebar item (e.g. Reviews)
* Verify you are prompted that you have unsaved changes
* Hit Cancel
* (Note: Changing the store location does not require Save - it always saves immediately)

* Navigate to Store > Settings > Taxes
* If your store has the TaxJar plugin enabled, disable it
* Change one of the tax settings
* Click on a sidebar item (e.g. Reviews)
* Verify you are prompted that you have unsaved changes
* Hit Cancel

* Put the tax setting back the way it was
* Click on a sidebar item (e.g. Reviews)
* Verify you are NOT prompted that you have unsaved changes
```

Looks like this:

<img width="1129" alt="screen shot 2017-11-22 at 12 23 59 pm" src="https://user-images.githubusercontent.com/1595739/33148332-52ee22d6-cf80-11e7-9e30-c7c4e61d89a2.png">

Note: The payments approach involves testing the non-emptiness of the actions list to determine if the payment settings form has unsaved changes. The taxes approach involves testing the initial state against the state to which the controls are bound to determine if the tax settings form has unsaved changes.
